### ISSUE TITLE
More control over invalidation of the custom functions metadata cache

### ISSUE THREAD

**Issue by wh1t3cAt1k** [OfficeDev/Excel-Custom-Functions#249]
Our product is in active development, and we often make changes to custom functions metadata, mostly non-breaking ones, like adding a new **optional** positional parameter at the end.

Based on https://github.com/OfficeDev/Excel-Custom-Functions/issues/205 , I understand that Office caches information about custom functions. 

I must admit that having no control over such caching is not convenient when doing pre-release testing, as the modified functions won't work after switching & clean-rebuilding from the production branch to the staging branch: even though we only added a new optional parameter at the end and did not edit the file (so that the optional parameter is not assigned to), the old function signature is cached, at least according to Excel IntelliSense. 

For some reason, in this situation we are seeing non-descriptive #VALUE errors all over the place.

_I know there is a workaround_ often suggested by MSFT specialists here to clear Office wef cache, which works: however, as I understand, the customers are impacted in a similar way by caching inconveniences.

It would be good to provide some sort of content-hashing mechanism (similar to what Webpack uses https://webpack.js.org/guides/caching/ ) that would force the invalidate the custom functions metadata when its shape changes.

**Comment by akrantz** [comment]
When you make changes, do you increment the version number of the add-in?

**Comment by wh1t3cAt1k** [comment]
@akrantz thanks for your response! No, will that force the re-invalidation of the cache?

To be honest, we try not to fiddle with the manifest at all if we can help it, because of certain issues outlined [here](https://camerondwyer.com/2018/09/11/office-add-in-manifest-updates-deployment-timing-and-potential-url-issues/):

> When users acquire your add-in a copy of the manifest.xml file is cached onto the host device and is only periodically resynced if the manifest.xml file is subsequently updated in AppSource/Office Store.  I have seen this take weeks on some combinations of host products/devices.

So it would be really great if we could signalize to invalidate the custom functions cache somehow otherwise.

**Comment by akrantz** [comment]
I don't know definitively but my recollection is that this can help. I think the cache uses the manifest version as part of the key, so incrementing it will invalidate it. The cache is not related to webpack, but rather that the custom function html, js, and metadata.json files are downloaded and cached for the add-in by Excel.

**Comment by wh1t3cAt1k** [comment]
Related issue for add-in HTML outlined on StackOverflow long ago: 

https://stackoverflow.com/questions/34168603/office-add-ins-caching-my-html-file

In a recent release, I set up NGINX to send revalidation `cache-control` headers for HTML, JS, and metadata JSON, to ensure the client always has the freshest version. 

I am not 100% certain it's going to help, though, could someone from the team clarify?

**TLDR: I wish there was a setting in the manifest to prevent automatic html / asset / metadata caching** (or at least force revalidation?) so that it could be controlled more reliably, and allow us to apply hotfixes without re-submitting the app.

Failing that, I would like to confirm if possible, that caching mechanisms will respect the `cache-control:` headers I added to remedy the issue, across the platforms (Mac, Excel Windows w/ IE11, Excel Windows w/ Edge Chromium, Excel Online on Edge Chromium) and for both the assets and custom functions metadata...
